FlaskSite/controllers/auth_controller.py

```diff
@@ -15,13 +15,48 @@ def register(username, password, email):
 
 def login(username, password):
     try:
-        (valid, token) = auth_service.login(username, password)
+        (valid, token, refresh_token) = auth_service.login(username, password)
         if not valid:
             return (
                 jsonify({"message": "Invalid username or password", "token": ""}),
                 200,
             )
-
-        return jsonify({"message": "Login successful", "token": token}), 200
+        # Create a response object
+        response = jsonify({"message": "Login successful", "access_token": token})
+        secure_cookie = current_app.config['SECURE_COOKIE']
+        # Set the refresh token as a secure cookie should be: response.set_cookie('refresh_token', refresh_token, httponly=True, secure=secure_cookie, samesite='None', path='/')
+        if secure_cookie:
+                response.set_cookie(
+                'refresh_token',
+                refresh_token,
+                httponly=True,
+                secure=secure_cookie,  # Enforces HTTPS in production
+                samesite='None' if secure_cookie else 'Lax',  # Use 'None' for cross-site requests in production
+                path='/'
+            )
+            
+        response.set_cookie('refresh_token', refresh_token)
+        return response, 200
     except Exception as e:
         return jsonify({"message": f"An error occurred {e}"}), 500
+
+
+
+def refresh_access_token(refresh_token):
+    try:
+        success, result = auth_service.refresh_access_token(refresh_token)
+        if not success:
+            if result == "Token expired":
+                # Clear the expired refresh token in the response
+                response = jsonify({"message": "Refresh token expired"})
+                response.status_code = 401
+                response.delete_cookie('refresh_token')  # Optional: clear cookie if used
+                return response
+            else:
+                return jsonify({"message": result}), 401  # Other token-related errors
```

Comment:
either modify refresh_access_token to not return str(e) in result or don't put exception in message